integration-tests/src/test/java/tachyon/master/JournalCrashIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import tachyon.Constants;
+import tachyon.TachyonURI;
+import tachyon.client.TachyonFS;
+import tachyon.conf.TachyonConf;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.InvalidPathException;
+import tachyon.thrift.TableDoesNotExistException;
+import tachyon.util.CommonUtils;
+
+/**
+ * Test master journal for crashes. Assert that test can replay the editlog and reproduce the
+ * correct state. Test both the single master(tachyon) and multi masters(tachyon-ft) mode.
+ */
+public class JournalCrashIntegrationTest {
+
+  /**
+   * Hold a client and keep creating files/tables.
+   */
+  class ClientThread implements Runnable {
+    /** The number of successfully created files/tables. */
+    public int mSuccessNum = 0;
```

Comment:
We should almost never use public member variable, please use a getter to access this value.